cgi-bin/LJ/Talk.pm

```diff
@@ -1830,7 +1831,8 @@ sub talkform {
 
             $ret .= "<input type='hidden' name='cookieuser' value='$remote->{'user'}' id='cookieuser' />\n";
             if ($screening eq 'A' ||
-                ($screening eq 'F' && !$journalu->trusts_or_has_member( $remote ))) {
+                ($screening eq 'F' && !$journalu->trusts_or_has_member( $remote )) ||
+                ( $journalu->has_autoscreen( $remote ) )) {
```

Comment:
1662 and 1775 are for OpenID, yes, but they (both of them) are the path that is "you're not currently logged in, this is what'd happen if you logged in with OpenID". As such, we don't actually have data on "who they'd be logged in as".